replicatelogs.go

```diff
@@ -503,11 +503,26 @@ func (v *VerifiedReplica) replicateVerifiedContext(
 
 	// if the remote and local are the same, we are done, provided the roots still match
 	if len(local.Data) == len(remote.Data) {
-		if !bytes.Equal(local.ConsistentRoots, remote.ConsistentRoots) {
+		if !verifiedStateEqual(local, remote) {
 			return fmt.Errorf("%w: %s, massif=%d", ErrRemoteLogInconsistentRootState, tenantIdentity, massifIndex)
 		}
 		return nil
 	}
 
 	return v.localReader.ReplaceVerifiedContext(remote, v.writeOpener)
 }
+
+func verifiedStateEqual(a *massifs.VerifiedContext, b *massifs.VerifiedContext) bool {
+	if len(a.Data) != len(b.Data) {
```

Comment:
Minor: Aren't we checking this again on line 506?

Reply:
Uh yeah. I can imporove that. The reason for the elevation is to avoid writing to disk when we know the remote is unchanged.

Reply:
I think I just figured the redundant check was a defence against future refactorings